test/cfa/ntp_conf_test.rb

```diff
@@ -131,88 +187,98 @@ def ntp_disk_file
 
   describe "#value" do
     it "obtains the value of the record" do
-      expect(record.value).to eq(augeas_element[:value].value)
+      record = ntp.records.first
+      expect(record.value).to eq("0.pool.ntp.org")
     end
   end
 
   describe "#value=" do
     it "sets the value of the record" do
-      value = "1.pool.ntp.org"
+      value = "3.pool.ntp.org"
+      record = ntp.records.first
       record.value = value
       expect(record.value).to eq(value)
+      ntp.save
+      expect(file.content).to include("server 3.pool.ntp.org iburst\n")
     end
   end
 
   describe "#comment" do
     context "when the record has not comment" do
       it "obtains nil" do
+        record = ntp.records.first
         expect(record.comment).to eq(nil)
       end
     end
 
     context "when the record has comment" do
-      let(:augeas_options) do
-        tree = CFA::AugeasTree.new
-        tree.add("#comment", "sample comment")
-        tree
-      end
-
       it "obtains the comment" do
-        expect(record.comment).to eq("sample comment")
+        record = ntp.records.to_a[1]
+        expect(record.comment).to eq("#sample comment")
       end
     end
   end
 
   describe "#comment=" do
     it "sets a comment to the record" do
-      comment = "sample comment"
+      comment = "#sample comment"
+      record = ntp.records.first
       record.comment = comment
       expect(record.comment).to eq(comment)
+      ntp.save
+      expect(file.content).to include("server 0.pool.ntp.org iburst#sample comment\n")
     end
   end
 
   describe "#==" do
     it "returns true for equal records" do
-      equal_record = described_class.new_from_augeas(augeas_element)
-      expect(record == equal_record).to be(true)
+      record = ntp.records.first
+      expect(record == record.dup).to be(true)
     end
 
     it "returns false for different records" do
-      different_record = CFA::NtpConf::Record.new(key: "server[]")
-      different_record.value = "10.10.10.10"
-      expect(record == different_record).to be(false)
+      records = ntp.records.to_a
+      expect(records[0] == records[1]).to be(false)
     end
   end
 
   describe "#type" do
     it "obtains the type of the record" do
-      expect(record.type).to eq("server")
+      expect(ntp.records.first.type).to eq("server")
     end
   end
 
   describe "#raw_options" do
     it "obtains options as string" do
-      expect(record.raw_options).to eq("iburst")
+      expect(ntp.records.first.raw_options).to eq("iburst")
     end
   end
 
   describe "#raw_options=" do
     it "sets options from a string" do
+      record = ntp.records.first
       record.raw_options = "iburst prefer"
       expect(record.options).to eq(["iburst", "prefer"])
+      ntp.save
+      expect(file.content).to include("server 0.pool.ntp.org iburst prefer\n")
```

Comment:
:+1: now it is much more readable and verificable that we really write what we want.